x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/LimitedRole.java

```diff
@@ -187,7 +182,6 @@ public boolean checkRunAs(String runAs) {
      */
     public static LimitedRole createLimitedRole(Role fromRole, Role limitedByRole) {
         Objects.requireNonNull(limitedByRole, "limited by role is required to create limited role");
-        return new LimitedRole(fromRole.names(), fromRole.cluster(), fromRole.indices(), fromRole.application(), fromRole.runAs(),
-                limitedByRole);
+        return new LimitedRole(fromRole.cluster(), fromRole.indices(), fromRole.application(), fromRole.runAs(), limitedByRole);
```

Comment:
I understand `fromRole.names()` is close to non-sense. But I wonder whether it is marginally justifiable to still keep them around just in case? I can be easily persuaded. :)

Reply:
I wouldn't clutter the code in any way for it. When/if we need that we can make the change.

Reply:
Ok sounds good as long as you do have an argument for it. Thanks!